lib/IDE/CompileInstance.cpp

```diff
@@ -325,7 +325,8 @@ bool CompileInstance::performCompile(
   if (CI->getDiags().hadAnyError())
```

Comment:
Would it make sense to also check before `performSema`? That would catch cases where multiple compile request are queued up and a request is cancelled before it had a chance to get scheduled.

Reply:
Good idea. 👍